Automattic/wp-calypso
# SMP: add Site Preview Link Modal on actions tooltip

- closes https://github.com/Automattic/dotcom-forge/issues/1197

#### Proposed Changes

* Add new action tooltip for `Coming soon`sites
* Add a modal showing the preview site toggle to generate the link

**Screencast**


https://user-images.githubusercontent.com/779993/205129307-66337c20-bc57-457f-a9aa-1e678963dd5e.mp4

**Screenshot**

`Share site for preview` action tooltip:

<img width="1725" alt="Screenshot 2022-11-25 at 18 21 34" src="https://user-images.githubusercontent.com/779993/204038902-f5e8da27-9a40-420a-af14-ce306b8cf155.png">


Modal:

<img width="1303" alt="Screenshot 2022-11-25 at 18 21 56" src="https://user-images.githubusercontent.com/779993/204038841-2cc3d77b-c00e-4deb-b8c6-a81591d4cc0c.png">


#### Testing Instructions



* Go to `/sites`
* Filter your sites by `Coming soon`.
* Click the three dots to see the actions menu on a site with Business or eCommerce plan.
* Click on `Share site for preview`.
* Observe a new modal is open.
* Click on the toggle to generate a new link. Copy the link.
* Open a new tab in Incognito mode and paste the copied link.
* Observe the site is visible using that link.


#### Pre-merge Checklist





- Blocked by:
   - https://github.com/Automattic/wp-calypso/pull/70242
   - https://github.com/Automattic/dotcom-forge/issues/1187
